src/wallet/db.h

```diff
@@ -115,13 +117,22 @@ class BerkeleyDatabase
         nUpdateCounter(0), nLastSeen(0), nLastFlushed(0), nLastWalletUpdate(0)
     {
         env = GetWalletEnv(wallet_path, strFile);
+        auto inserted = env->m_databases.emplace(strFile, std::ref(*this));
+        assert(inserted.second);
         if (mock) {
             env->Close();
             env->Reset();
             env->MakeMock();
         }
     }
 
+    ~BerkeleyDatabase() {
```

Comment:
nit, could add `final` to class since this is not virtual.

Reply:
`final` can only be added to virtual method.

Reply:
@ken2812221 I mean `class BerkeleyDatabase final`.

Reply:
Change would be unrelated to this PR, and IMO, it's better only to use final when you have virtual methods and actual optimizations this would allow or bugs it would prevent. There's a core guideline that touches on this: http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-final

Reply:
Nice thanks.